sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogCosmosSDKClient.scala

```diff
@@ -169,17 +169,39 @@ private[spark] case class CosmosCatalogCosmosSDKClient(cosmosAsyncClient: Cosmos
 
     private def getPartitionKeyDefinition(containerProperties: Map[String, String]): PartitionKeyDefinition = {
         val partitionKeyPath = CosmosContainerProperties.getPartitionKeyPath(containerProperties)
-
         val partitionKeyDef = new PartitionKeyDefinition
         val paths = new util.ArrayList[String]
-        paths.add(partitionKeyPath)
+        val pathList = partitionKeyPath.split(",").toList
+        if (pathList.size >= 2) {
+            partitionKeyDef.setKind(CosmosContainerProperties.getPartitionKeyKind(containerProperties) match {
+                case Some(pkKind) => {
+                    if (pkKind == PartitionKind.HASH.toString) {
+                        throw new IllegalArgumentException("PartitionKind HASH is not supported for multi-hash partition key")
+                    }
+                    PartitionKind.MULTI_HASH
+                }
+                case None => PartitionKind.MULTI_HASH
+            })
+            partitionKeyDef.setVersion(CosmosContainerProperties.getPartitionKeyVersion(containerProperties) match {
+                case Some(pkVersion) =>
+                    {
+                        if (pkVersion == PartitionKeyDefinitionVersion.V1.toString) {
+                            throw new IllegalArgumentException("PartitionKeyVersion V1 is not supported for multi-hash partition key")
+                        }
+                        PartitionKeyDefinitionVersion.V2
+                    }
+                case None => PartitionKeyDefinitionVersion.V2
+            })
+            pathList.foreach(path => paths.add(path.trim))
+        } else {
+            partitionKeyDef.setKind(PartitionKind.HASH)
+            paths.add(partitionKeyPath)
```

Comment:
maybe move line 178-181 logic to here? as for multi-hash, the version has already been configured.

Reply:
Ok, added the logic hash to there, and also moved set version so it is not done redundantly in the case of sub-partition.